Sources/Moya/Response.swift

```diff
@@ -116,4 +116,25 @@ public extension Response {
             return string
         }
     }
+
+    /// Maps data received from the signal into a Decodable object.
+    ///
+    /// - parameter atKeyPath: Optional key path at which to parse object.
+    /// - parameter using: A `JSONDecoder` instance which is used to decode data to an object.
+    func map<D: Decodable>(_ type: D.Type, atKeyPath keyPath: String? = nil, using decoder: JSONDecoder = JSONDecoder()) throws -> D {
+        let jsonData: Data
+        if let keyPath = keyPath {
+            guard let jsonDictionary = (try mapJSON() as? NSDictionary)?.value(forKeyPath: keyPath) as? [String: Any] else {
+                throw MoyaError.jsonMapping(self)
+            }
+            jsonData = try JSONSerialization.data(withJSONObject: jsonDictionary, options: [])
```

Comment:
don't we wanna catch the error here as well and make it a `.jsonMapping` as well? 🤔

Reply:
Great!